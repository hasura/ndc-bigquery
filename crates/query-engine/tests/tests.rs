mod common;

#[test]
fn it_converts_select_with_limit() {
    let result = common::test_translation("select_with_limit").unwrap();
    insta::assert_snapshot!(result);
}

#[test]
fn it_select_where_string() {
    let result = common::test_translation("select_where_string").unwrap();
    insta::assert_snapshot!(result);
}

#[test]
fn it_aggregate_count_albums() {
    let result = common::test_translation("aggregate_count_albums").unwrap();
    insta::assert_snapshot!(result);
}

#[test]
fn it_aggregate_distinct_albums() {
    let result = common::test_translation("aggregate_distinct_albums").unwrap();
    insta::assert_snapshot!(result);
}

#[test]
fn it_aggregate_function_albums() {
    let result = common::test_translation("aggregate_function_albums").unwrap();
    insta::assert_snapshot!(result);
}
