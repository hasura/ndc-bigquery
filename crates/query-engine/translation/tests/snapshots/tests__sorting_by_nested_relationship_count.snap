---
source: crates/query-engine/tests/tests.rs
expression: result
---
SELECT
  row_to_json("universe") AS "universe"
FROM
  (
    SELECT
      *
    FROM
      (
        SELECT
          coalesce(json_agg(row_to_json("rows")), '[]') AS "rows"
        FROM
          (
            SELECT
              "Artist"."Name" AS "Name"
            FROM
              "public"."Artist" AS "Artist"
              LEFT OUTER JOIN LATERAL (
                SELECT
                  count("%ORDER_PATH_PART_0_Album"."AlbumId") AS "AlbumId"
                FROM
                  (
                    SELECT
                      "%ORDER_PATH_PART_0_Album"."AlbumId" AS "AlbumId"
                    FROM
                      "public"."Album" AS "%ORDER_PATH_PART_0_Album"
                    WHERE
                      (
                        true
                        AND (
                          "Artist"."ArtistId" = "%ORDER_PATH_PART_0_Album"."ArtistId"
                        )
                      )
                  ) AS "%ORDER_PATH_PART_0_Album"
              ) AS "%ORDER_0_FOR_Artist" ON ('true')
            ORDER BY
              "%ORDER_0_FOR_Artist"."AlbumId" DESC
            LIMIT
              3
          ) AS "rows"
      ) AS "rows"
  ) AS "universe"

[]
